server/data/data-api/src/main/java/org/apache/james/droplists/api/DropListEntry.java

```diff
@@ -0,0 +1,122 @@
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.droplists.api;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import com.google.common.base.Preconditions;
+
+public class DropListEntry {
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private Optional<OwnerScope> ownerScope = Optional.empty();
+        private Optional<String> owner = Optional.empty();
+        private Optional<String> deniedEntityType = Optional.empty();
+        private Optional<String> deniedEntity = Optional.empty();
+
+        public Builder ownerScope(OwnerScope ownerScope) {
+            Preconditions.checkNotNull(ownerScope);
+            this.ownerScope = Optional.of(ownerScope);
+            return this;
+        }
+
+        public Builder owner(String owner) {
+            Preconditions.checkNotNull(owner);
+            this.owner = Optional.of(owner);
+            return this;
+        }
+
+        public Builder deniedEntityType(String deniedEntityType) {
+            Preconditions.checkNotNull(deniedEntityType);
+            this.deniedEntityType = Optional.of(deniedEntityType);
+            return this;
+        }
+
+        public Builder deniedEntity(String deniedEntity) {
+            Preconditions.checkNotNull(deniedEntity);
+            this.deniedEntity = Optional.of(deniedEntity);
+            return this;
+        }
+
+        public Builder copy(DropListEntry dropListEntry) {
+            this.ownerScope = dropListEntry.getOwnerScope();
+            this.owner = dropListEntry.getOwner();
+            this.deniedEntityType = dropListEntry.getDeniedEntityType();
+            this.deniedEntity = dropListEntry.getDeniedEntity();
+            return this;
+        }
+
+        public DropListEntry build() {
+            return new DropListEntry(ownerScope, owner, deniedEntityType, deniedEntity);
+        }
+    }
+
+    private Optional<OwnerScope> ownerScope;
+    private Optional<String> owner;
+    private Optional<String> deniedEntityType;
+    private Optional<String> deniedEntity;
```

Comment:
Aren't those all mandatory?

Can we really have a drop list entry without a target entity?